packages/core/src/particle/ParticleGenerator.ts

```diff
@@ -764,4 +766,150 @@ export class ParticleGenerator {
     out.push(vertexBufferBinding);
     return index;
   }
+
+  /** @internal */
+  _calculateLocalBoundingBox(): void {
+    const { min, max } = this._renderer._localBounds;
+    const { _tempVector30: directionMax, _tempVector31: directionMin } = ParticleGenerator;
+    const lifetime = this.main.startLifetime._getMinMaxValue({ min: 0, max: 0 }).max;
+
+    // StartSpeed's impact
+    if (this.emission.shape) {
+      this.emission.shape._getStartPositionRange({ min, max });
+      this.emission.shape._getDirectionRange({ min: directionMin, max: directionMax });
+    } else {
+      min.set(0, 0, 0);
+      max.set(0, 0, 0);
+      directionMin.set(0, 0, -1);
+      directionMax.set(0, 0, 0);
+    }
+    this._adjustBoundingBoxFromDirectionalVelocity(directionMax, directionMin, lifetime, this.main.startSpeed);
+
+    // GravityModifier's impact
+    const { _tempQuat0: worldInvQuat, _tempVector32: direction } = ParticleGenerator;
+    Quaternion.invert(this._renderer.entity.transform.worldRotationQuaternion, worldInvQuat);
+    // Transform gravity direction into local space
+    direction.copyFrom(this._renderer.scene.physics.gravity);
+    Vector3.transformByQuat(direction, worldInvQuat, direction);
+
+    this._adjustBoundingBoxFromDirectionalVelocity(
+      direction,
+      direction,
+      0.5 * lifetime * lifetime,
+      this.main.gravityModifier
+    );
+
+    // StartSize's impact
+    const maxSize = this.main.startSize3D
+      ? Math.max(
+          this.main.startSizeX._getMinMaxValue({ min: 0, max: 0 }).max,
+          this.main.startSizeY._getMinMaxValue({ min: 0, max: 0 }).max,
+          this.main.startSizeZ._getMinMaxValue({ min: 0, max: 0 }).max
+        )
+      : this.main.startSize._getMinMaxValue({ min: 0, max: 0 }).max;
+
+    min.x -= maxSize;
+    max.x += maxSize;
+
+    min.y -= maxSize;
+    max.y += maxSize;
+
+    min.z -= maxSize;
+    max.z += maxSize;
+
+    // VelocityOverLifetime Module's impact
+    if (this.velocityOverLifetime.enabled) {
+      const { velocityX, velocityY, velocityZ } = this.velocityOverLifetime;
+      directionMin.set(
+        velocityX._getMinMaxValue({ min: 0, max: 0 }).min,
+        velocityY._getMinMaxValue({ min: 0, max: 0 }).min,
+        velocityZ._getMinMaxValue({ min: 0, max: 0 }).min
+      );
+      directionMax.set(
+        velocityX._getMinMaxValue({ min: 0, max: 0 }).max,
+        velocityY._getMinMaxValue({ min: 0, max: 0 }).max,
+        velocityZ._getMinMaxValue({ min: 0, max: 0 }).max
+      );
+
+      if ((this.velocityOverLifetime.space = ParticleSimulationSpace.World)) {
+        Vector3.transformByQuat(directionMin, worldInvQuat, directionMin);
+        Vector3.transformByQuat(directionMax, worldInvQuat, directionMax);
+      }
+
+      this._adjustBoundingBoxFromDirectionalVelocity(directionMin, directionMax, lifetime);
+    }
+
+    // SimulationSpace's Impact
+    if (this.main.simulationSpace === ParticleSimulationSpace.World) {
```

Comment:
Temp code?

Reply:
its's a temp solution.

Reply:
now it goes with the firstElement etc. pointers to get real-time bounding boxes